### PR TITLE
Azure Logic App PowerShell Cmdlets

### PR DESCRIPTION
**Pull request by pankajsn**
This pull request contains the Azure Logic App PowerShell Cmdlets implementation which introduces following cmdlets:

Stop-AzureLogicAppRun
Get-AzureLogicAppAccessKey
Get-AzureLogicApp
Get-AzureLogicAppRunAction
Get-AzureLogicAppRunHistory
Get-AzureLogicAppTrigger
Get-AzureLogicAppTriggerHistory
New-AzureLogicApp
Remove-AzureLogicApp
Start-AzureLogicApp
Set-AzureLogicAppAccessKey
Set-AzureLogicApp

Also, includes: Test Automation, Installer Setup Update, Help Documentation, ps1xml

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@pankajsn__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (pankajsn). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by hovsepm** [comment]
@azuresdkci add to whitelist

**Comment by pankajsn** [comment]
azurecmdfiles.wxi file got messed up during merge. File is fixed and committed now.

**Comment by pankajsn** [comment]
AzurePowerShell.Test.targets updated for test integration.
Updated Readme.md and ChangeLog.md with cmdlets description.

File: README.md [hunk 1]
```diff
@@ -119,6 +119,11 @@ This repository contains a set of PowerShell cmdlets for developers and administ
 * Azure Automation
   * Manage automation accounts
   * Manage automation jobs, runbooks and schedules
+* Logic Apps
```
Comment (markcowl) [inline README.md]: Please remove, we are no longer updating the README for new RPs

File: src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1 [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+DotNetFrameworkVersion = '4.0'  
+  
+# Minimum version of the common language runtime (CLR) required by this module  
+CLRVersion='4.0'  
+  
+# Processor architecture (None, X86, Amd64, IA64) required by this module  
+ProcessorArchitecture = 'None'  
+  
+# Modules that must be imported into the global environment prior to importing this module  
+RequiredModules = @( @{ ModuleName = 'AzureRM.Profile'; ModuleVersion = '1.0.1'} )
```
Comment (markcowl) [inline src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1]: Update to the latest (currently 1.0.4)

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+    <Reference Include="System" />
+    <Reference Include="System.Core" />
+    <Reference Include="System.Net" />
+    <Reference Include="System.Net.Http" />
+    <Reference Include="System.Xml" />
+    <Reference Include="System.Xml.Linq" />
+    <Reference Include="System.Core">
+      <RequiredTargetFramework>3.5</RequiredTargetFramework>
+    </Reference>
+    <Reference Include="System.Management.Automation, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (markcowl) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj]: two references to system.management.automation?  Please remove the previous

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure.Authentication">
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.Authentication.1.2.1-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.Authentication.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure.TestFramework, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <SpecificVersion>False</SpecificVersion>
+      <HintPath>..\..\..\packages\Microsoft.Rest.ClientRuntime.Azure.TestFramework.1.2.0-preview\lib\net45\Microsoft.Rest.ClientRuntime.Azure.TestFramework.dll</HintPath>
+    </Reference>
+    <Reference Include="Microsoft.WindowsAzure.Management, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\packages\Microsoft.WindowsAzure.Management.4.1.1\lib\net40\Microsoft.WindowsAzure.Management.dll</HintPath>
```
Comment (markcowl) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj]: Why do you need this?  There is no way you could possibly use it. Please remove

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+    <Compile Include="ScenarioTests\WorkflowAccessKeyTests.cs" />
+    <Compile Include="ScenarioTests\WorkflowController.cs" />
+    <Compile Include="ScenarioTests\WorkflowRunTests.cs" />
+    <Compile Include="ScenarioTests\WorkflowTests.cs" />
+    <Compile Include="ScenarioTests\WorkflowTriggerTests.cs" />
+    <Compile Include="UnitTests\CreateLogicAppTests.cs" />
+    <Compile Include="UnitTests\LogicAppUnitTestBase.cs" />
+  </ItemGroup>
+  <ItemGroup>
+    <None Include="MSSharedLibKey.snk" />
```
Comment (markcowl) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj]: should not need this in a test project

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+using System.Runtime.InteropServices;
+using Xunit;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("Microsoft.Azure.Commands.LogicApp.Test")]
+[assembly: AssemblyDescription("")]
+[assembly: AssemblyConfiguration("")]
+[assembly: AssemblyCompany("")]
```
Comment (markcowl) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs]: Please fill out as per other assemblies

File: README.md (hunk 1 above)
Comment (pankajsn) [inline README.md]: [Fixed] : Section removed from the Readme.md

File: src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1 (hunk 2 above)
Comment (pankajsn) [inline src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1]: [Fixed] : Profile Module Version is 1.0.3. Fixed

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj (hunk 4 above)
Comment (pankajsn) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj]: [Fixed] : This is was not being used. Removed the unused reference.

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj (hunk 5 above)
Comment (pankajsn) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj]: [No Fix] : I also noticed this one but this is consistent across all the projects.

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs (hunk 6 above)
Comment (pankajsn) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs]: [Fixed] : Added Description and Company.

File: ChangeLog.md [hunk 7]
```diff
@@ -1,5 +1,18 @@
...
+  * Get-AzureLogicAppRunAction
+  * Get-AzureLogicAppRunHistory
+  * Get-AzureLogicAppTrigger
+  * Get-AzureLogicAppTriggerHistory
+  * New-AzureLogicApp
+  * Remove-AzureLogicApp
+  * Start-AzureLogicApp
+  * Set-AzureLogicAppAccessKey
+  * Set-AzureLogicApp
+  * Stop-AzureLogicAppRun
```
Comment (hovsepm) [inline ChangeLog.md]: Your changes will not go out as part of February release. Please add a new tag 
## 2016.03.08 version 1.3.0 
and then add your change log text.

File: src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1 [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+DotNetFrameworkVersion = '4.0'  
+  
+# Minimum version of the common language runtime (CLR) required by this module  
+CLRVersion='4.0'  
+  
+# Processor architecture (None, X86, Amd64, IA64) required by this module  
+ProcessorArchitecture = 'None'  
+  
+# Modules that must be imported into the global environment prior to importing this module  
+RequiredModules = @( @{ ModuleName = 'AzureRM.Profile'; ModuleVersion = '1.0.3'} )
```
Comment (hovsepm) [inline src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1]: AzureRM.Profile module would be version 1.0.4 after current release. Please update it to 1.0.4.

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj (hunk 5 above)
Comment (hovsepm) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj]: Please remove - we will delete it from other projects as well.

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyVersion)]
+[assembly: AssemblyFileVersion(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyFileVersion)]
```
Comment (hovsepm) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs]: your module will not be a part of RDFE package. Please have the same version string here as you have in the src/ResourceManager/LogicApp/AzureRM.LogicApp.psd1 file.

**Comment by hovsepm** [comment]
@pankajsn  all the cmdlet names should follow [Verb]-AzureRM[prefix][noun] naming. Please fix.

File: ChangeLog.md (hunk 7 above)
Comment (pankajsn) [inline ChangeLog.md]: [Fixed]: Moved to new section as suggested. Below tag indicates next release in March. I just want to know, What is the process if we want to release these changes early?

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj (hunk 5 above)
Comment (pankajsn) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Commands.LogicApp.Test.csproj]: [Fixed]

File: src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs (hunk 9 above)
Comment (pankajsn) [inline src/ResourceManager/LogicApp/Commands.LogicApp.Test/Properties/AssemblyInfo.cs]: [Fixed]

**Comment by pankajsn** [comment]
[Fixed]

Get-AzureLogicAppAccessKey    >>           Get-AzureRmLogicAppAccessKey
Get-AzureLogicApp                         >>           Get-AzureRmLogicApp
Get-AzureLogicAppRunAction    >>           Get-AzureRmLogicAppRunAction
Get-AzureLogicAppRunHistory   >>           Get-AzureRmLogicAppRunHistory
Get-AzureLogicAppTrigger           >>           Get-AzureRmLogicAppTrigger
New-AzureLogicApp                       >>           New-AzureRmLogicApp
Remove-AzureLogicApp                >>           Remove-AzureRmLogicApp
Start-AzureLogicApp                      >>           Start-AzureRmLogicApp
Set-AzureLogicAppAccessKey     >>           Set-AzureRmLogicAppAccessKey
Set-AzureLogicApp                          >>           Set-AzureRmLogicApp
Stop-AzureLogicAppRun                >>           Stop-AzureRmLogicAppRun
Get-AzureLogicAppTriggerHistory            >>           Get-AzureRmLogicAppTriggerHistory

**Comment by pcgeek86** [comment]
@pankajsn @hovsepm Thanks for ensuring naming standards. Are we also ensuring that there is detailed cmdlet help before they are released, please?

Cheers,
Trevor Sullivan

**Comment by pankajsn** [comment]
Yes, Detailed cmdlet help is in place.

Regards,
~Pankaj

Sent from my Windows Phone
________________________________

**Comment by hovsepm** [comment]
@pankajsn  Please pull from upstream dev and resolve conflicts.

**Pull request merged**